src/core/stream/period/period_stream.ts

```diff
@@ -114,12 +115,13 @@ export default function PeriodStream({
   wantedBufferAhead$,
 } : IPeriodStreamArguments) : Observable<IPeriodStreamEvent> {
   const { period } = content;
+  const needsBufferFlushing$ = new ReplaySubject<void>(1);
```

Comment:
Sure!

Reply:
Here you tell how it is used which can be very quickly checked by reading how it is used in the file.
It would be more helpful to comment on why this is necessary.